spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestIcebergExpressions.java

```diff
@@ -71,4 +72,34 @@ public void testTruncateExpressions() {
         ImmutableList.of(row(100, 10000L, new BigDecimal("10.50"), "10", "12")),
         sql("SELECT int_c, long_c, dec_c, str_c, CAST(binary_c AS STRING) FROM v"));
   }
+
+  @Test
+  public void testBucketExpressions() {
```

Comment:
Could you please add another unit test described in issue #7682? Basically for a table with binary bucket partition.